File: convert_icons.js
/**
 * 将SVG图标转换为PNG格式
 * 需要安装: npm install sharp
 */

const fs = require('fs');
const path = require('path');

// 检查是否安装了sharp
let sharp;
try {
  sharp = require('sharp');
} catch (e) {
  console.log('请先安装依赖: npm install sharp');
  console.log('或者使用在线工具转换SVG到PNG: https://convertio.co/zh/svg-png/');
  process.exit(1);
}

async function convertSvgToPng(svgPath, pngPath, width, height) {
  try {
    await sharp(svgPath)
      .resize(width, height)
      .png()
      .toFile(pngPath);
    console.log(`✓ 已转换: ${svgPath} -> ${pngPath}`);
    return true;
  } catch (error) {
    console.log(`✗ 转换失败 ${svgPath}: ${error.message}`);
    return false;
  }
}

async function main() {
  console.log('开始转换图标...\n');

  // tabBar图标转换 (81x81)
  const tabbarDir = 'images/tabbar';
  const tabbarIcons = [
    { svg: 'home.svg', png: 'home.png' },
    { svg: 'home-active.svg', png: 'home-active.png' },
    { svg: 'handshake.svg', png: 'handshake.png' },
    { svg: 'handshake-active.svg', png: 'handshake-active.png' },
    { svg: 'publish.svg', png: 'publish.png' },
    { svg: 'publish-active.svg', png: 'publish-active.png' },
    { svg: 'member.svg', png: 'member.png' },
    { svg: 'member-active.svg', png: 'member-active.png' },
    { svg: 'user.svg', png: 'user.png' },
    { svg: 'user-active.svg', png: 'user-active.png' }
  ];

  console.log('转换tabBar图标 (81x81)...');
  for (const icon of tabbarIcons) {
    const svgPath = path.join(tabbarDir, icon.svg);
    const pngPath = path.join(tabbarDir, icon.png);
    if (fs.existsSync(svgPath)) {
      await convertSvgToPng(svgPath, pngPath, 81, 81);
    }
  }

  console.log('\n所有图标转换完成！');
  console.log('如果转换失败，请使用在线工具: https://convertio.co/zh/svg-png/');
}

main().catch(console.error);

File: convert_icons.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
将SVG图标转换为PNG格式
需要安装: pip install cairosvg pillow
"""

import os
import sys

try:
    from cairosvg import svg2png
except ImportError:
    print("请先安装依赖: pip install cairosvg")
    sys.exit(1)

def convert_svg_to_png(svg_path, png_path, width=81, height=81):
    """将SVG转换为PNG"""
    try:
        svg2png(url=svg_path, write_to=png_path, output_width=width, output_height=height)
        print(f"✓ 已转换: {svg_path} -> {png_path}")
        return True
    except Exception as e:
        print(f"✗ 转换失败 {svg_path}: {e}")
        return False

def main():
    # tabBar图标转换 (81x81)
    tabbar_dir = "images/tabbar"
    tabbar_icons = [
        "home.svg", "home-active.svg",
        "handshake.svg", "handshake-active.svg",
        "publish.svg", "publish-active.svg",
        "member.svg", "member-active.svg",
        "user.svg", "user-active.svg"
    ]
    
    print("开始转换tabBar图标...")
    for icon in tabbar_icons:
        svg_path = os.path.join(tabbar_dir, icon)
        png_path = os.path.join(tabbar_dir, icon.replace(".svg", ".png"))
        if os.path.exists(svg_path):
            convert_svg_to_png(svg_path, png_path, 81, 81)
    
    # 其他图标转换 (48x48)
    icons_dir = "images/icons"
    icons = [
        "chart.svg", "shop.svg", "handshake.svg", "pie-chart.svg", "users.svg",
        "search.svg", "fire.svg", "trend.svg", "new.svg", "education.svg",
        "beauty.svg", "food.svg", "lightning.svg"
    ]
    
    print("\n开始转换其他图标...")
    for icon in icons:
        svg_path = os.path.join(icons_dir, icon)
        png_path = os.path.join(icons_dir, icon.replace(".svg", ".png"))
        if os.path.exists(svg_path):
            convert_svg_to_png(svg_path, png_path, 48, 48)
    
    # LOGO转换 (200x80)
    logo_svg = "images/logo.svg"
    logo_png = "images/logo.png"
    if os.path.exists(logo_svg):
        print("\n开始转换LOGO...")
        convert_svg_to_png(logo_svg, logo_png, 200, 80)
    
    print("\n所有图标转换完成！")

if __name__ == "__main__":
    main()
